src/common/utils.ts

```diff
@@ -105,20 +106,35 @@ function xrpToDrops(xrp: BigNumber.Value): string {
     .toString(10)
 }
 
-function toRippledAmount(amount: Amount): RippledAmount {
+function toRippledAmount(amount: Amount | RippledAmount): RippledAmount {
+  if (typeof amount === 'string')
+    return amount;
+
   if (amount.currency === 'XRP') {
     return xrpToDrops(amount.value)
   }
   if (amount.currency === 'drops') {
     return amount.value
   }
+
+  let issuer = amount.counterparty
+    ? amount.counterparty
+    : amount.issuer
+    ? amount.issuer
+    : undefined
+
+  if (isValidXAddress(issuer)) {
+    const {classicAddress, tag} = xAddressToClassicAddress(issuer)
+
+    if (tag)
+      throw new RippledError("Issuer cannot have tag")
```

Comment:
I'm not 100% sure that would work... How would this handle issuers that were classic addresses (which is most cases)? 

I think we'd have to handle them in the `catch` block, which I don't love..

Reply:
Maybe something like:

```js
let issuer =  amount.counterparty || amount.issuer

if (!validClassicAddress(issuer)) {
    const {classicAddress, tag} = xAddressToClassicAddress(issuer)  // This can throw an error all the way to the user
    
    if (tag)
         throw new ValidationError("Issuer X-address includes a tag") 
         
    issuer = classicAddress
}
```

Reply:
No, I think my suggestion would work as-is. If `issuer` is a classic address, then no handling is needed. It should be left alone, which is what my suggestion does (`issuer` is unchanged).

Reply:
Ahh, hence the empty catch block... 

Per usual, I am wrong 😉